algorithms/benches/snark/varuna.rs

```diff
@@ -66,13 +67,15 @@ fn snark_prove(c: &mut Criterion) {
 
         let max_degree = AHPForR1CS::<Fr, VarunaHidingMode>::max_degree(1000, 1000, 1000).unwrap();
         let universal_srs = VarunaInst::universal_setup(max_degree).unwrap();
-        let universal_prover = &universal_srs.to_universal_prover().unwrap();
         let fs_parameters = FS::sample_parameters();
+        let hiding_bound = AHPForR1CS::<Fr, VarunaHidingMode>::zk_bound().unwrap_or(0);
 
         let (circuit, _) = TestCircuit::gen_rand(mul_depth, num_constraints, num_variables, rng);
 
-        let params = VarunaInst::circuit_setup(&universal_srs, &circuit).unwrap();
-        b.iter(|| VarunaInst::prove(universal_prover, &fs_parameters, &params.0, &circuit, rng).unwrap())
+        let (pk, _) = VarunaInst::circuit_setup(&universal_srs, &circuit).unwrap();
+        let degree_info = pk.circuit.index_info.degree_info::<Fr, VarunaHidingMode>();
```

Comment:
One small but significant consideration here is that if we support Wasm targets for the web (which I think at some point may constitute one of the most common places where people run Varuna proofs) the parameter downloads need to be fetched using asynchronous methods (they'll fail if called from a synchronous trim method).

This degree info is the key piece of information needed to download parameters, so I'd like to propose that either the proving key has an async method for fetching these parameters (like an async version of `to_commiter_key` or a method like `preload_parameters`) OR at the very least, that the degree info be available so that the caller can handle the concern loading these parameters into the `UniversalProver`